acceptance-tests/src/main/java/io/blueocean/ath/SmartWebElement.java

```diff
@@ -271,20 +198,24 @@ public String getText() {
     }
 
     @Override
-    public List<WebElement> findElements(By by) {
-        WebElement e = getElement();
-        return e.findElements(by);
+    public List<WebElement> findElements(By locator) {
+        return buildWait().until((WebDriver driver) -> {
+            List<WebElement> elements = getElement().findElements(locator);
+            return LocalDriver.wrapElements(driver, elements, locator);
+        });
     }
 
     @Override
-    public WebElement findElement(By by) {
-        WebElement e = getElement();
-        return e.findElement(by);
+    public WebElement findElement(By locator) {
+        return buildWait().until((WebDriver driver) -> {
+            WebElement matchedElement = getElement().findElement(locator);
+            return LocalDriver.wrapElement(getDriver(), matchedElement, locator);
+        });
     }
 
     @Override
     public boolean isDisplayed() {
-        WebElement e = getElement();
+        WebElement e = buildWait().until(ExpectedConditions.visibilityOf(element));
         return e.isDisplayed();
```

Comment:
I think in its current state this method can only ever return true or throw an exception. I might be wrong. Not sure if this should have some error handling baked in, or maybe left alone.